### PR TITLE
remove the auto pub get behavior

### PR DESCRIPTION
**Pull request by devoncarew**
Take another stab at https://github.com/dart-lang/dartdoc/pull/2894 - see if I can address the test failures.

### CODE REVIEW

**Comment by srawlins** [comment]
Love this idea. I've been slowly making fewer tests depend on `pub get`, but I'm sure most still do.

**Comment by devoncarew** [comment]
> Love this idea. I've been slowly making fewer tests depend on `pub get`, but I'm sure most still do.

Thanks! I'm making a bit pf progress on this PR - will update the latest to see which tests still fail.

**Comment by devoncarew** [comment]
Updated the PR to fix one test failure I saw locally - creating a package_config.json file w/ more than one package reference (https://github.com/dart-lang/dartdoc/pull/2930/commits/d9fdded21d79bf3d351354e6f1cf779ae1568585).

File: test/src/test_descriptor_utils.dart [hunk 1]
```diff
@@ -21,8 +23,32 @@ Future<d.DirectoryDescriptor> createPackage(
...
+  "name": "$packageName",
+  "rootUri": "../",
+  "packageUri": "lib/",
+  "languageVersion": "$languageVersion"
+}''');
+  if (parsedYaml.containsKey('dependencies')) {
+    final dependencies = parsedYaml['dependencies'] as Map;
+    for (var dep in dependencies.keys) {
+      // This only accepts 'path' deps.
+      final depConfig = dependencies[dep] as Map;
```
Comment (srawlins) [inline test/src/test_descriptor_utils.dart:41]: Woohoo very nice

**Review by srawlins — APPROVED** [review]

File: test/src/test_descriptor_utils.dart (hunk 1 above)
Comment (devoncarew) [reply test/src/test_descriptor_utils.dart:41]: Thanks! From memory, this was just needed to get one test passing, that was testing doc references between two test packages.

**Comment by devoncarew** [comment]
@srawlins - the last failure isn't immediately clear to me:

```
test-dartdoc-flutter-plugin
flutter-docs-flutter-plugin: Documenting test_package_flutter_plugin...
flutter-docs-flutter-plugin: Initialized dartdoc with 39 libraries in 25.2 seconds
flutter-docs-flutter-plugin: Generating docs for library testlib from file:///home/runner/work/dartdoc/dartdoc/testing/flutter_packages/test_package_flutter_plugin/lib/testlib.dart...
flutter-docs-flutter-plugin: Validating docs...
flutter-docs-flutter-plugin: no issues found
flutter-docs-flutter-plugin: Documented 1 public library in 0.9 seconds
flutter-docs-flutter-plugin: Success! Docs generated into /tmp/test_package_flutter_pluginFMBGSU

Instance of 'GrindTestFailure'
#0      expectFileContains (file:///home/runner/work/dartdoc/dartdoc/tool/grind.dart:35:7)
#1      testDartdocFlutterPlugin (file:///home/runner/work/dartdoc/dartdoc/tool/grind.dart:1223:3)
<asynchronous suspension>
#2      _RootZone.bindUnaryCallbackGuarded.<anon> (dart:async/zone.dart:1657:12)
```

I'm not sure if it is an actual issue or not. Do you have time to take over the PR / land as it?

**Comment by devoncarew** [comment]
I remain fairly stumped by the last failure (which I can repro locally by running `dart run grinder validate-flutter-docs`). Perhaps there some implicit dependency on state created by earlier grind tasks? Some other place(s) we have to run `pub get`?

**Comment by devoncarew** [comment]
I _believe_ I've not gotten this passing. I had to run `dart pub get`, from the dart sdk vended in with the flutter sdk, in the `testing/flutter_packages/test_package_flutter_plugin` directory, as part of the flutter grind test script.

**Comment by srawlins** [comment]
Fantastic!

**Comment by devoncarew** [comment]
I see that you have dartdoc `5.0.0` staged; I'm not sure if you were thinking of merging this in, but this might make 5.0.0 potentially more breaking than you were intending - I'm happy to have this shipped separately.

**Pull request merged**